<?xml version="1.0"?>
<launch>

    <env name="GAZEBO_MODEL_PATH" 
        value="$(find rmi)/models:$(find rmi)/models/hanoi" />

  <arg name="gui" default="true" doc="Start gazebo gui" />

  <arg name="moveit" default="true" doc="Use MoveIt for motion planning"/>
  <arg name="rviz"   default="true" doc="Launch RViz with movit planner plugin"/>
  <arg name="paused" default="false" doc="Start simulation paused" />
  <arg name="world_name" default="$(find rmi)/worlds/hanoi.world"/>
  <arg name="limited" default="true"  
      doc="Use the description in limited mode (Every axis rotates from -PI to PI)"/>

  <!-- startup simulated world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" default="$(arg world_name)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>

  <!-- send robot URDF to param server -->
  <include file="$(find ur_e_description)/launch/ur10e_upload.launch">
    <arg name="limited" value="$(arg limited)"/>
  </include>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_robot" pkg="gazebo_ros" type="spawn_model" 
      args="-urdf -param robot_description -model robot -z 0.7" 
      respawn="false" output="screen" />

  <!-- spawn Hanoi rod support in gazebo -->
  <node name="spawn_fixture" pkg="gazebo_ros" type="spawn_model" 
      args="-sdf -database hanoi_rod_support -model support -x 0 -y 0.7 -z 0.7" 
      respawn="false" output="screen" />

  <include file="$(find ur_e_gazebo)/launch/controller_utils.launch"/>

  <!-- start this controller -->
  <rosparam file="$(find ur_e_gazebo)/controller/arm_controller_ur10e.yaml" command="load"/>
  <node name="arm_controller_spawner" pkg="controller_manager" type="controller_manager" 
      args="spawn arm_controller gripper_controller" respawn="false" output="screen"/>

  <!-- load other controllers -->
  <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" 
      respawn="false" output="screen" args="load joint_group_position_controller"/>

  <!-- start a fake weiss gripper -->
  <node name="fake_gripper" pkg="ur_e_gazebo" type="fake_wr-cr200-85.py" 
      respawn="true" output="log"/>

  <!-- launch MoveIt! stuff -->
  <group if="$(arg moveit)">
    <include file="$(find ur10_e_moveit_config)/launch/ur10_e_moveit_planning_execution.launch">
      <arg name="limited" value="$(arg limited)"/>
      <arg name="sim"     value="true"/>
    </include>
    <include if="$(arg rviz)" file="$(find ur10_e_moveit_config)/launch/moveit_rviz.launch">
      <arg name="config" value="true"/>
    </include>
  </group>

</launch>

